### ISSUE TITLE
Trix field - attaching files issue on registering a Nova resource

### ISSUE THREAD

**Issue by thisprojectworks** [laravel/nova-issues#473]
Hi!

Laravel version: 5.7
Laravel Nova version: 1.0.12

I have a problem with the TRIX field when attaching files.
I have added the migrations and in the Kernel file as it says in the documentation to upload files with Trix.

2018_09_05_143756_create_trix_uploads.php File:

```
<?php

use Illuminate\Support\Facades\Schema;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Database\Migrations\Migration;

class CreateTrixUploads extends Migration
{
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        Schema::create('nova_pending_trix_attachments', function (Blueprint $table) {
            $table->increments('id');
            $table->string('draft_id')->index();
            $table->string('attachment');
            $table->string('disk');
            $table->timestamps();
        });

        Schema::create('nova_trix_attachments', function (Blueprint $table) {
            $table->increments('id');
            $table->string('attachable_type');
            $table->unsignedInteger('attachable_id');
            $table->string('attachment');
            $table->string('disk');
            $table->string('url')->index();
            $table->timestamps();

            $table->index(['attachable_type', 'attachable_id']);
        });
    }

    /**
     * Reverse the migrations.
     *
     * @return void
     */
    public function down()
    {
        Schema::dropIfExists('nova_trix_attachments');
        Schema::dropIfExists('nova_pending_trix_attachments');
    }
}
```

I describe the problem:

I have a Nova resource (App/Nova /Exhibitions) with this code: 

```
<?php

namespace App\Nova;

use Laravel\Nova\Fields\Date;
use Laravel\Nova\Fields\HasMany;
use Laravel\Nova\Fields\ID;
use Illuminate\Http\Request;
use Laravel\Nova\Fields\Trix;
use Laravel\Nova\Http\Requests\NovaRequest;
use Laravel\Nova\Fields\Markdown;
use Laravel\Nova\Fields\Text;
use Laravel\Nova\Fields\Boolean;
use Laravel\Nova\Panel;
use Laravel\Nova\Fields\BelongsToMany;

class Exhibition extends Resource
{
    /**
     * The model the resource corresponds to.
     *
     * @var string
     */
    public static $model = 'App\Exhibition';

    /**
     * The single value that should be used to represent the resource when being displayed.
     *
     * @var string
     */
    public static $title = 'id';

    /**
     * The columns that should be searched.
     *
     * @var array
     */
    public static $search = [
        'id',
    ];

    /**
     * Get the fields displayed by the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function fields(Request $request)
    {
        return [
            ID::make()->sortable(),

            Trix::make('Images')->withFiles('public'),

            Text::make('Nombre', 'name')
                ->sortable()
                ->rules('required', 'max:400'),

            new Panel('Español', $this->spanishFields()),

            new Panel('English', $this->englishFields()),

            Date::make('Fecha de Inicio', 'date_start')
                ->sortable()
                ->rules('required', 'date'),

            Date::make('Fecha de Fin', 'date_end')
                ->sortable()
                ->rules('required', 'date'),

            Text::make('Hashtag')
                ->hideFromIndex()
                ->sortable()
                ->rules('max:200'),


            BelongsToMany::make('Artists'),

            HasMany::make('Presses'),


            Boolean::make('Activo', 'active'),
        ];
    }

    /**
     * Get the spanish fields for the resource.
     *
     * @return array
     */
    protected function spanishFields()
    {
        return [
            Markdown::make('Texto - español', 'text_es')
                ->sortable()
        ];
    }


    /**
     * Get the english fields for the resource.
     *
     * @return array
     */
    protected function englishFields()
    {
        return [
            Markdown::make('Texto - english', 'text_en')
                ->sortable()
        ];
    }

    /**
     * Get the cards available for the request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function cards(Request $request)
    {
        return [];
    }

    /**
     * Get the filters available for the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function filters(Request $request)
    {
        return [];
    }

    /**
     * Get the lenses available for the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function lenses(Request $request)
    {
        return [];
    }

    /**
     * Get the actions available for the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function actions(Request $request)
    {
        return [];
    }
}

```

We  added the TRIX field in the resource, when I'm going to register a new one (App/Nova/Exhibitions) I add the photos that I want in the field, but when I click the create button I gives this error:
```
SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'attachable_id' cannot be null (SQL: insert into `nova_trix_attachments` (`attachable_type`, `attachable_id`, `attachment`, `disk`, `url`, `updated_at`, `created_at`) values (App\Exhibition, , cHeZaaVv2LlyNwgrbIJhJvUmzcluD5TDx4bOb3VV.png, public, http://espaitactel.test/storage/cHeZaaVv2LlyNwgrbIJhJvUmzcluD5TDx4bOb3VV.png, 2018-09-06 10:32:39, 2018-09-06 10:32:39))
```
However I have observed that when it comes to an (App/Nova/Exhibitions) already registered and I edit it then I do not get an error and I can save it.

This is due to the fact that when registering, it is not adding me the model ID (App/Nova/Exhibitions) since it does not yet have it. I think it's something that you should solve.

Thank you very much!
Carlos

**Comment by weeman1337** [comment]
Same here, Nova 1.0.13

**Comment by lotestudio** [comment]
Same, Nova 1.0.15!

**Comment by ModernArtisan** [comment]
Yep, now it doesn't work on resource create. It should persist drafts on `Model::created` event if model not exists yet.

**Comment by JapSeyz** [comment]
Still apparent in Nova  1.0.19

**Comment by MalkSof** [comment]
Same Nova 1.1.4
![laravel nova](https://user-images.githubusercontent.com/42155335/46551485-473f5e00-c8ae-11e8-8203-8b4ffa148edb.jpg)

**Comment by timvandijck** [comment]
Also experiencing this.

**Comment by frankperez87** [comment]
Seems like it works for me if you try and upload images on when updating an existing record. Just not when you're creating new records.

**Comment by iHazzam** [comment]
Same issue 1.1.4 need a fix asap please

**Comment by robmpreston** [comment]
Having this issues as well.

**Comment by taylorotwell** [comment]
Share the entire stack trace. You may just need to make the column nullable and it could be a documentation issue.

**Comment by jeverington** [comment]
Yep that worked for me. Thanks @taylorotwell. I guess this might need to be updated in the docs.
`$table->unsignedInteger('attachable_id')->nullable();`

**Comment by rosscturner** [comment]
@taylorotwell solves it in the short term for me but it also leaves an orphaned row in the DB when creating a new record. Will this be addressed or should we write an additional clean up script?

**Comment by taylorotwell** [comment]
Should be fixed on 1.1.5

**Comment by taylorotwell** [comment]
Nullable should no longer be needed.
